pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -298,6 +299,36 @@ def test_iloc_setitem_bool_indexer(self, klass):
         expected = DataFrame({"flag": ["x", "y", "z"], "value": [2, 3, 4]})
         tm.assert_frame_equal(df, expected)
 
+    def test_setitem_complete_columns_different_dtypes(self):
+        # GH: 20635
+        df = DataFrame({"A": ["a", "b"], "B": ["1", "2"], "C": ["3", "4"]})
+        df.loc[:, ["B", "C"]] = df.loc[:, ["B", "C"]].astype("int64")
+        expected = DataFrame({"A": ["a", "b"], "B": [1, 2], "C": [3, 4]}, dtype="int64")
+        tm.assert_frame_equal(df, expected)
+
+    def test_setitem_single_column_as_series_different_dtype(self):
```

Comment:
same comment about (add to the original frame) and test Int64

Reply:
Done